docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -1681,6 +1683,61 @@ public class ItemResource {
 ----
 <1> The template path for this method will be `items/item-and-order`.
 
+[[type_safe_fragments]]
+==== Type-safe Fragments
+
+You can also define a type-safe <<fragments,fragment>> in your Java code.
+A fragment method is annotated with `@io.quarkus.qute.CheckedFragment`.
+The name of the fragment is derived from the annotated method name. 
+The part before the last occurence of a dollar sign `$` is the method name of the related type-safe template.
+The part after the last occurence of a dollar sign is the fragment identifier.
+The strategy defined by the relevant `CheckedTemplate#defaultName()` is honored when constructing the defaulted names.
+
+.Type-safe Fragment Example
+[source,java]
+----
+import io.quarkus.qute.CheckedFragment;
+import io.quarkus.qute.CheckedTemplate;
+
+@CheckedTemplate
+class Templates {
+
+  // defines a type-safe template
+  static native TemplateInstance items(List<Item> items);
+
+  // defines a fragment of Templates#items() with identifier "item"
+  @CheckedFragment
+  static native TemplateInstance items$item(Item item); <1>
```

Comment:
So following the example below, this method should be named `items$item_li`?

Reply:
Good catch! Yes, we should either change the method name to `items$item_li(Item item)` or change the fragment id to `item`.